src/gt4py/next/common.py

```diff
@@ -499,8 +499,12 @@ def replace(self, index: int | Dimension, *named_ranges: NamedRange) -> Domain:
         if index < 0:
             index += len(self.dims)
         new_dims, new_ranges = zip(*named_ranges) if len(named_ranges) > 0 else ((), ())
-        dims = self.dims[:index] + new_dims + self.dims[index + 1 :]
-        ranges = self.ranges[:index] + new_ranges + self.ranges[index + 1 :]
+        if new_dims == self.dims:
+            dims = new_dims
+            ranges = new_ranges
```

Comment:
because the else branch would output the second dimension twice, i.e. say that `new_dims` is:
```(Dimension(value='IDim', kind=<DimensionKind.HORIZONTAL: 'horizontal'>), Dimension(value='KDim', kind=<DimensionKind.VERTICAL: 'vertical'>))```

at index 0 in the else branch this would become: 
```(Dimension(value='IDim', kind=<DimensionKind.HORIZONTAL: 'horizontal'>), Dimension(value='KDim', kind=<DimensionKind.VERTICAL: 'vertical'>), Dimension(value='KDim', kind=<DimensionKind.VERTICAL: 'vertical'>))```

Reply:
I don't know if it's useful, but that is probably intended. What would you do if `self.dims` == `reverse(new_dims)`?